pkg/tools/etcd_helper.go

```diff
@@ -32,21 +33,33 @@ import (
 	"github.com/golang/glog"
 )
 
+const (
+	DefaultEtcdPathPrefix = "/registry"
+)
+
 // EtcdHelper offers common object marshalling/unmarshalling operations on an etcd client.
 type EtcdHelper struct {
 	Client EtcdGetSet
 	Codec  runtime.Codec
 	// optional, no atomic operations can be performed without this interface
 	Versioner EtcdVersioner
+	// optional prefix for all etcd keys
+	PathPrefix string
 }
 
 // NewEtcdHelper creates a helper that works against objects that use the internal
 // Kubernetes API objects.
-func NewEtcdHelper(client EtcdGetSet, codec runtime.Codec) EtcdHelper {
+func NewEtcdHelper(client EtcdGetSet, codec runtime.Codec, prefix string) EtcdHelper {
+	if prefix == "" {
+		prefix = DefaultEtcdPathPrefix
+	} else {
+		prefix = path.Join("/", prefix, DefaultEtcdPathPrefix)
+	}
 	return EtcdHelper{
-		Client:    client,
-		Codec:     codec,
-		Versioner: APIObjectVersioner{},
+		Client:     client,
+		Codec:      codec,
+		Versioner:  APIObjectVersioner{},
+		PathPrefix: prefix,
```

Comment:
Please modify the helper functions in this package to prepend keys with this prefix.